syntax = "proto3";

package netservice;

service NetService {
  rpc OperationService(OperationRequest) returns (OperationResponse) {}
}

message OperationRequest {
  fixed32 identification = 1; 

  enum Operation {
    // Interal operations. Do not use.
    ServerGet = 0;
    ServerPut = 1; // WAL-Write
    CompactionRead = 2;
    CompactionWrite = 3;
    FlushWrite = 4;

    // Operations that can be used by the client
    Get = 5;
    Put = 6;
    Delete = 7;
    // Not tracked by the server
    BatchPut = 8;
  }
  Operation operation = 2;
  fixed32 total_length = 3;
  uint64 sequence_number = 4;
  repeated bytes keys = 5;
  repeated string values = 6;
}

message OperationResponse {
  uint64 sequence_number = 1;
  string result = 2;
  string get_result = 3;
}
